dspace-api/src/main/java/org/dspace/discovery/IndexClient.java

```diff
@@ -26,132 +23,53 @@
 import org.dspace.core.Constants;
 import org.dspace.core.Context;
 import org.dspace.handle.factory.HandleServiceFactory;
-import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.scripts.DSpaceRunnable;
+import org.springframework.beans.factory.annotation.Autowired;
 
-/**
- * Class used to reindex dspace communities/collections/items into discovery
- *
- * @author Kevin Van de Velde (kevin at atmire dot com)
- * @author Mark Diggory (markd at atmire dot com)
- * @author Ben Bosman (ben at atmire dot com)
- */
-public class IndexClient {
-
-    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger(IndexClient.class);
-
-    /**
-     * Default constructor
-     */
-    private IndexClient() { }
-
-    /**
-     * When invoked as a command-line tool, creates, updates, removes content
-     * from the whole index
-     *
-     * @param args the command-line arguments, none used
-     * @throws SQLException           An exception that provides information on a database access error or other errors.
-     * @throws IOException            A general class of exceptions produced by failed or interrupted I/O operations.
-     * @throws SearchServiceException if something went wrong with querying the solr server
-     */
-    public static void main(String[] args) throws SQLException, IOException, SearchServiceException {
-
-        Context context = new Context(Context.Mode.READ_ONLY);
-        context.turnOffAuthorisationSystem();
-
-        String usage = "org.dspace.discovery.IndexClient [-cbhf] | [-r <handle>] | [-i <handle>] or nothing to " +
-                "update/clean an existing index.";
-        Options options = new Options();
-        HelpFormatter formatter = new HelpFormatter();
-        CommandLine line = null;
-
-        options.addOption(OptionBuilder
-                              .withArgName("handle to remove")
-                              .hasArg(true)
-                              .withDescription(
-                                  "remove an Item, Collection or Community from index based on its handle")
-                              .create("r"));
+public class IndexClient extends DSpaceRunnable {
```

Comment:
This Class is missing Javadocs.  It looks like they were removed...when they likely should have been updated.